docs/rules/no-new-buffer.md

```diff
@@ -0,0 +1,27 @@
+# Enforce the use of `Buffer.from()` and `Buffer.alloc()` instead of the deprecated `new Buffer()`
+
+Enforces the use of [Buffer.from](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_array) and [Buffer.alloc()](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_alloc_size_fill_encoding) instead of [new Buffer()](https://nodejs.org/api/buffer.html#buffer_new_buffer_array), which has been deprecated since Node.js 4.
+
+
+## Fail
+
+```js
+const buf = new Buffer('7468697320697320612074c3a97374', 'hex');
+const buf = new Buffer([0x62, 0x75, 0x66, 0x66, 0x65, 0x72]);
+```
+
+```js
```

Comment:
Good question, I wanted to make a difference between `new Buffer` being converted to `Buffer.from` and the `new Buffer` being converted to `Buffer.alloc` as you can see in the `Pass` section below. To keep those blocks consistent, I splitted them up. I could merge them though, whatever you guys prefer.